feat: configurable `sentiment_classfier` for `CounterFactualMetrics` (#138)

Mihir3 opened the pull request:
## Description
<!--- Provide a general summary of your changes. -->
Added support for specifying an optional `sentiment_classifier` type as a parameter from users when they call the [`CounterFactualMetrics.evaluate`](https://github.com/cvs-health/langfair/blob/main/langfair/metrics/counterfactual/counterfactual.py#L62) function.

Proposed Solution in this PR: 

- [`CounterfactualMetrics`](https://github.com/cvs-health/langfair/blob/main/langfair/metrics/counterfactual/counterfactual.py#L38) class: Modified the `evaluate()` function and added support for passing `sentiment_classifier` as a classifier parameter to `SentinmentBias` class.
- Re-arranged logic code in `evaluate()` function for clarity.

Supported Sample Usage:
```
from langfair.metrics.counterfactual import CounterfactualMetrics
cm = CounterfactualMetrics()
cf_result1 = cm.evaluate(texts1, texts2, attribute="gender", sentiment_classifier="vader")
cf_result2 = cm.evaluate(texts1, texts2, attribute="gender", sentiment_classifier="roberta")
```
<!--- Mention related issues, pull requests, or discussions with #<issue/PR/discussion ID>. -->
Resolves Issue: (#138)
<!--- Tag people for whom this PR may be of interest using @<username>. -->

## Contributor License Agreement
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [x] confirm you have signed the [LangFair CLA](https://forms.office.com/pages/responsepage.aspx?id=uGG7-v46dU65NKR_eCuM1xbiih2MIwxBuRvO0D_wqVFUMlFIVFdYVFozN1BJVjVBRUdMUUY5UU9QRS4u&route=shorturl)

## Tests
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

- Added another test `test_CounterfactualMetrics_ClassifierSupport` similar to `test_CounterfactualMetrics`. The proposed logic passes all tests.

## Documentation
<!--- Select all that apply by putting an x between the brackets: [x] -->
- [ ] no documentation changes needed
- [ ] README updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

@dylanbouchard I'd like to request your insights, on whether any API documentation should be updated to reflect support for this feature — and if so, which parts specifically.

## Screenshots
<!--- If applicable, please add screenshots. -->

dylanbouchard commented on tests/test_counterfactual_metrics.py:
```diff
@@ -27,11 +27,11 @@
 )
 
-datafile_path = "tests/data/counterfactual/counterfactual_data_file.json"
+datafile_path = "data/counterfactual/counterfactual_data_file.json"
```
Looks like this path edit is causing the unit tests to fail. Please revert to the old path

Mihir3 replied:
Fixed it.

dylanbouchard commented on tests/test_counterfactual_metrics.py:
```diff
@@ -27,11 +27,11 @@
     data = json.load(f)
 
-actual_result_file_path = "tests/data/counterfactual/counterfactual_results_file.json"
+actual_result_file_path = "data/counterfactual/counterfactual_results_file.json"
```
Same comment about path.

Mihir3 replied:
Fixed it.

dylanbouchard reviewed (changes requested):
Please update paths in unit tests and I will rerun. Thank you!

dylanbouchard commented on langfair/metrics/counterfactual/counterfactual.py:
```diff
@@ -65,6 +65,7 @@ def evaluate(
         texts2: list,
         attribute: str = None,
         return_data: bool = False,
+        sentiment_classifier: Optional[str] = None,
```
Let's add the docstring for this one. It can be copied directly from `SentimentBias` class docstring

Mihir3 replied:
Done!

dylanbouchard reviewed (changes requested):
Look great! Let's implement these very minor changes:
1. Add docstring for new parameter
2. Remove print statement from `SentimentBias` class
3. Ruff formatting

Mihir3 commented:
Thank you for the comments, Dylan! I've updated the docstring and implemented ruff formatting.

https://github.com/cvs-health/langfair/blob/19cf6a1a58bcd908ef2860aaf2e05c7cad58c804/langfair/metrics/counterfactual/metrics/sentimentbias.py#L166
For the print statement in`SentimentBias` class, I didn’t modify that file — it already exists on the main branch. Would you still like me to remove it as part of this PR?

dylanbouchard reviewed (commented):
Looks good
